src/dependency-manager/src/configs/graph/index.ts

```diff
@@ -0,0 +1,262 @@
+/* eslint-disable no-empty */
+import { EnvironmentVault } from '../../environment-config/base';
+import VaultManager from '../../vault-manager';
+import { BaseDnsConfig, BaseEnvironmentConfig } from '../environment-config';
+import { EnvironmentBuilder } from '../environment.builder';
+import { BaseParameterValueConfig, BaseParameterValueFromConfig, BaseServiceConfig, BaseValueFromDependencyConfig, BaseValueFromVaultConfig } from '../service-config';
+import { ServiceBuilder } from '../service.builder';
+
+interface GraphEnrichmentOptions {
+  getConfig: (ref: string) => Promise<BaseServiceConfig>;
+  getRegistryImage: (ref: string) => string;
+  getHostAssignment: (config: BaseServiceConfig, interface_key: string) => Promise<string>;
+  getPortAssignment: (config: BaseServiceConfig, interface_key: string) => Promise<number>;
+  debug?: boolean;
+  skip_nodes?: string[];
+}
+
+class GraphNotBuiltError extends Error {
+  message = 'Please build and validate the graph before applying changes';
+}
+
+class ValueFromDependencyError extends Error {
+  constructor(
+    service_name: string,
+    dependency_name: string,
+    parameter_key: string
+  ) {
+    super();
+    this.message = `${service_name} is trying to populate the parameter, ${parameter_key}, from the service ${dependency_name}, but it was not installed as a dependency.`;
+  }
+}
+
+export class EnvironmentGraph {
+  private _is_built = false;
+  private _base_config: BaseEnvironmentConfig;
+  private _enriched_config: BaseEnvironmentConfig;
+
+  constructor(config: BaseEnvironmentConfig) {
+    this._base_config = config;
+    this._enriched_config = EnvironmentBuilder.create();
+  }
+
+  private async loadParametersForService(
```

Comment:
Need to add back support for Vault and prob need to have hook like getConfig to read from files on local filesystem for cli.